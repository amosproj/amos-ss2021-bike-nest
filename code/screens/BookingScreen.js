import React, { useEffect } from 'react';
import { Pressable, StyleSheet, Text, TextInput, View, Image} from 'react-native';
import { Dimensions } from "react-native";
import Colors from '../styles/Colors';
import BikeNest_NavigationFooter from '../components/BikeNest_NavigationFooter';
import { mainStyles } from "../styles/MainStyles";
import ModalDropdown from 'react-native-modal-dropdown';
import colors from '../styles/Colors';
import { BikenestService } from "../services/BikenestService";
import { Alert } from 'react-native';

var width = Dimensions.get('window').width; //full width
var height = Dimensions.get('window').height; //full height

export default function BookingScreen({ route, navigation }) {
  let bikenestService = new BikenestService();

  const [textSlots, setTextSlots] = React.useState();
  const [textHours, setTextHours] = React.useState();
  const [textEbike, setTextEbike] = React.useState();
  const [costSlots, setcostSlots] = React.useState();
  const [costHours, setcostHours] = React.useState();
  const [costEbike, setcostEbike] = React.useState();
  const [estimatedPrice, setEstimatedPrice] = React.useState();

  const [spotsLeft, setSpotsLeft] = React.useState(0);
  const [chargingOption, setChargingOption] = React.useState(true);
  const [bikenestName, setbikenestName] = React.useState('default Bikenest');

  let bikenest = route.params.state;
  console.log(bikenest.id);

  const fetchBikenestInfos = () => {
      bikenestService.getBikenestInfo(bikenest.id).then((response) => {
        setSpotsLeft(response.currentSpots);
        setChargingOption(response.chargingAvailable);
        setbikenestName(response.name);
      }).catch((error) => {
        console.error("getBikenest Infos Error:" + error);
     });
  };

  const onPressInfo = () => {
    //zurück zu Find Bike Nest
    alert('this is an info.');
  };
  const onPressOrder = () => {
    //weiter zu order verarbeitung
    if(textSlots != null && textHours != null && textEbike != null){
      navigation.navigate("Payment", {state: bikenest, name: bikenestName, slots: textSlots, time: textHours, ebike: textEbike, price: estimatedPrice});
    } else {
      Alert.alert("Halt", "Bitte wähle Slots, Stunden und Ebike-Option aus")
    } 
  }
  const getLocation = () => {
    fetchBikenestInfos();
    return bikenestName;
  };
  const getSelectedHours = (value) => {
    if(value == 0){
      setTextHours('1 Tag');
      setcostHours(1);
    }else if(value == 1){
      setTextHours('2 Tage');
      setcostHours(2);
    }else if(value == 2){
      setTextHours('3 Tage');
      setcostHours(3);
    }else if(value == 3){
      setTextHours('1 Monat');
      setcostHours(12);
    } else {
      setTextHours('Keine Zeit ausgewählt.');
      setcostHours(0);
    }
  }
  const getSelectedSlots = (value) => {

    if(value == 0){
      setTextSlots('1 Slot');
      setcostSlots(1);
    } else if(value == 1){
      setTextSlots('2 Slots');
      setcostSlots(2);
    } else {
      setTextSlots('Keine Slots ausgewählt.');
      setcostSlots(0);
    }
  }

  const getSelectedEbike = (value) => {
    if(value == 0){
      setTextEbike('1 E-Bike Ladestation');
      setcostEbike(5);
    } else if(value == 1){
      setTextEbike('2 E-Bike Ladestation');
      setcostEbike(10);
    } else if(value == 2){
      setTextEbike('keine E-Bike Ladestation');
      setcostEbike(0);
    } else {
      setTextEbike('kein E-bike ausgewählt.');
      setcostEbike(0);
    }
  }

  useEffect(()=>{
    let price = 0

    if(costHours != null){
      price = costHours;
    } 
    if (costHours != null && costSlots != null){
      price = (costHours*costSlots);
    } 
    if (costHours != null && costSlots != null && costEbike != null){
      price = (costHours*costSlots) + costEbike;
    }
    setEstimatedPrice(price);
     
  });
  return (
    <View style={myStyles.container}>
      <View style={myStyles.paymentContainer}>
        <View style={{alignSelf:'center'}}>
            <Text style={myStyles.h2}>
                Buchungsinformationen <Image onPress={() => onPressInfo()} source={require('../assets/payment/info.png')}/>
            </Text>
        </View>
        <View style={myStyles.headline}>
            <Text style={myStyles.h3}> Details </Text>
            <Text style={[myStyles.h3, {fontWeight: 'bold'}]} onPress={()=>navigation.navigate("FindBikeNest")}> Ändern </Text>
        </View>
        <View style={myStyles.headline}>
            <Text style={myStyles.stdText}>
              BIKE NEST {"\n"}
              {getLocation()}
            </Text>
        </View>
        <Image source={require('../assets/payment/Divider.png')} style={myStyles.divider}/>
        <View style={myStyles.headline}>
            <Text style={myStyles.h3}> {"\n"} Zusätzliche Informationen </Text>
        </View>
       
        <View style={myStyles.headline}>
          <Text style={myStyles.h3}> Wie viele Slots? </Text>
          <ModalDropdown onSelect={(value) => getSelectedSlots(value)} options={['1 Slot', '2 Slots']} defaultValue={'Bitte wählen...'} textStyle={{fontSize: 18 }} dropdownTextStyle={{fontSize: 18}}/>
        </View>
        <View style={myStyles.headline}>
          <Text style={myStyles.h3}> Wie lange buchen? </Text>
          <ModalDropdown onSelect={(value) => getSelectedHours(value)} options={['1 Tag', '2 Tage', '3 Tage', '1 Monat']} defaultValue={'Bitte wählen...'} textStyle={{fontSize: 18 }} dropdownTextStyle={{fontSize: 18}}/>
        </View>
        <View style={myStyles.headline}>
          <Text style={myStyles.h3}> E-Bike Ladestation? </Text>
          <ModalDropdown onSelect={(value) => getSelectedEbike(value)} options={['ja, eine', 'ja, zwei', 'nein, danke']} defaultValue={'Bitte wählen...'} textStyle={{fontSize: 18 }} dropdownTextStyle={{fontSize: 18}}/>
        </View>

        <Image source={require('../assets/payment/Divider.png')} style={myStyles.divider}/>
        <View style={myStyles.headline}>
          <Text style={myStyles.h3}> Zusammenfassung</Text>
        </View>
        <View style={myStyles.headline}>
          <Text style={myStyles.stdText}>
                {"\n"} 
                BIKE NEST {"\n"}
                {getLocation()} {"\n"}
                {textSlots}{"\n"}
                {textHours}{"\n"}
                {textEbike}{"\n"}
                ~{estimatedPrice}€
              </Text>
        </View>
          {/* <Pressable style={[myStyles.reserved, {justifyContent: 'flex-end'}]}  onPress={() => getPrice(this)}>
            <Text style={myStyles.h3}>Preis Kalkulieren</Text> 
          </Pressable> */}
          <Pressable style={[myStyles.reserved, {justifyContent: 'flex-end'}]}  onPress={() => onPressOrder(this)}>
            <Text style={myStyles.h3}>Weiter</Text> 
            <Image style={{margin: 10}} source={require('../assets/payment/mail-send.png')} />
          </Pressable>
      </View>
      <BikeNest_NavigationFooter/>
    </View>
  )
}

const myStyles = StyleSheet.create({
  container: {
      flex: 1,
      backgroundColor: '#ffffff',
  },
  paymentContainer: {
    flex: 1,
    padding: 15,
    marginTop: 30,
  },
  h2: {
      fontSize: 27,
      letterSpacing: 2,
      fontWeight: "400",
  },
  h3:{
      fontSize: 18,
      fontWeight: "300",
  },
  stdText:{
      fontSize: 14,
  },
  divider: {
      marginLeft: -15,
      marginTop: 10,
      marginBottom: 10,
      width: 500,
  },
  headline: {
      flex: 1,
      fontSize: 18,
      fontWeight: "300",
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center', 
  },
  reserved: {
      flexDirection: 'row',
      justifyContent: 'flex-start',
      alignItems: 'center', 
  },
  zahlungsmethode: {
      flexDirection: 'column',
      justifyContent: 'space-between',
      alignItems: 'center', 
  },
  button: {
      alignItems: 'center',
      justifyContent: 'center',
      elevation: 3,
      width: 328,
      height: 55,
      borderRadius: 38,
      margin: 9,
      backgroundColor: Colors.UI_Light_4
  },
  btnPress: {
    alignItems: 'center',
    justifyContent: 'center',
    elevation: 3,
    width: 328,
    height: 55,
    borderRadius: 38,
    margin: 9,
    backgroundColor: Colors.UI_Light_4,
    borderColor: Colors.UI_Light_3,
  },
  buttonContent: {
      flexDirection: 'row',
      justifyContent: 'space-evenly',
      width: '100%'
    },
  buttonImage: {
      alignSelf: 'center'
    },
  halfButton: {
      width: 150,
      height: 55,
      color: Colors.UI_Light_3,
      backgroundColor: Colors.UI_Light_4,
      borderRadius: 38,
      margin: 10,
      padding: 15,
    }
})
