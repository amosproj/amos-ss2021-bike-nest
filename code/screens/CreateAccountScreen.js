import React from 'react';
import { Text, View, ScrollView } from 'react-native';
import { CreateAccountManually } from '../components/CreateAccountManually';
import { CreateAccountVia3rdParty } from '../components/CreateAccountVia3rdParty';
import { mainStyles } from "../styles/MainStyles";


export function CreateAccountScreen() {
  return (
    <View style={mainStyles.container}>
      <ScrollView >

        <View style={[mainStyles.container, { marginTop: 40 }]}>
          <Text style={mainStyles.h1}>Profil erstellen</Text>
        </View>

        <CreateAccountVia3rdParty />

        <View style={mainStyles.container}>
          <Text style={mainStyles.stdText}>ODER MIT EMAIL ANMELDEN</Text>
        </View>

        <CreateAccountManually />
      </ScrollView>  
    </View>
  );
}