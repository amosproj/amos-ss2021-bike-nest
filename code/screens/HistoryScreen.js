import React, { useState } from 'react';
import { ImageBackground, Pressable, StyleSheet, Text, View, Image, TouchableOpacity  } from 'react-native';
import { Dimensions } from "react-native";
import Avatar from '../assets/Avatar.png'; 
import bike from '../assets/bike.png'; 
import { SimpleLineIcons } from '@expo/vector-icons';
import { AntDesign } from '@expo/vector-icons';
import global from '../components/GlobalVars';
import { mainStyles } from "../styles/MainStyles";
import BikeNest_NavigationFooter from '../components/BikeNest_NavigationFooter';


var width = Dimensions.get('window').width; //full width
var height = Dimensions.get('window').height; //full height


export default function HistoryScreen({ navigation }) {
  // const [myListData, setData] = useState("");

  let tryGETBooking = () => {
    console.log('start pulling reservation info');

    return fetch(global.globalIPAddress + "/booking/all", {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
        },
    })
        .then((response) => response.json())
        .then((json) => {
          alert(JSON.stringify(json));
          console.log(json);
        })
        .catch((error) => {
            console.error(error);
        });
}

  return (
    <View style={mainStyles.container}>
      <View style={styles.historyContainer}>
        <View style={styles.containerRow}>
          <Image source={Avatar} style={styles.avatar} />
          <Text style={styles.name} >
            Max Muster </Text>
        </View>
        <TouchableOpacity onPress={() => navigation.navigate("Lock") }
          style={styles.heightBike, {
            backgroundColor: '#FFF',
            height: 230,
            width: 360,
            margin: 10,
            position: 'relative'
          }}>
          <ImageBackground
            source={bike}
            style={{
              height: 230,
              width: 360,
              position: 'absolute',
            }}
          />
          <View style={{ flex: 1, alignItems: 'flex-start', justifyContent: 'flex-start', padding: 30}}>
            <Text style={{fontSize: 16}}>Vielen Dank für die Reservierung! </Text>
            <Text style={{fontSize: 16}}>Begib dich zu folgendem BIKE NEST: {"\n"}</Text>
            <Text style={{fontWeight: 'bold', fontSize: 16}}>BIKE NEST {"\n"}Nürnberg HBF </Text>
          </View>
        </TouchableOpacity>

      <TouchableOpacity onPress={() => navigation.navigate("EditPersonalInformation")} style={styles.buttonLock}>
        <Text style={styles.buttonLockOwner}> Max Muster's bike </Text>
        <Text style={styles.buttonLockText}> locked </Text>
      </TouchableOpacity>

      <View style={styles.containerRow}>
      <TouchableOpacity style={styles.time}>
        <Text style={styles.timeText}> Zeit verging </Text>
        <SimpleLineIcons name="clock" size={24} color="black" />
        <Text style={styles.timeRecord}> 1 Tag </Text> 
      </TouchableOpacity>

      <TouchableOpacity style={styles.cost}>
        <Text style={styles.costText}> Parkkosten </Text>
        <AntDesign name="creditcard" size={24} color="black" />
        <Text style={styles.costRecord}> 50 $ </Text> 
      </TouchableOpacity>
      </View>

      <TouchableOpacity onPress={() => tryGETBooking(this)} style={styles.buttonHistory}>
        <Text style={styles.buttonHistoryText}> Frühere Reservierungen und Zahlungen </Text>
      </TouchableOpacity>
      </View>
      <BikeNest_NavigationFooter></BikeNest_NavigationFooter>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    // backgroundColor: 'red',
    alignItems: 'center',
    justifyContent: 'center',
  },
  historyContainer: {
    flex: 1,
    padding: 15,
    marginTop: 20,
    margin: 10,
    justifyContent: 'space-evenly',
    position: 'relative',
  },
  containerRow:{
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    margin: 10,
  },
  avatar: {
    flex: 1,
    maxWidth: 60, 
    resizeMode: 'contain',
    marginLeft: 10,
    marginRight: 10,
  },
  name: {
    flex: 1,
    color: '#000000',
    fontSize: 18,
    fontWeight: 'bold',
    margin: 10,
  },
  welcome: {
    flex: 1,
    color: '#000000',
    fontSize: 15,
    fontWeight: 'normal',
  },
  heightBike: {
    height: 200,
    position: 'relative',           
  },
  textBike: {
    position: 'absolute',
    color: '#000000',
    fontSize:16,
    fontWeight:'bold',
    flex: 1,                    
    // backgroundColor: 'yellow',
  },
  place: {
    flex: 1,
    color: '#FFFFFF',
    fontSize:16,
    fontWeight:'bold',
  },
  nest: {
    flex: 1,
    color: '#FFFFFF',
    fontSize:14,
    fontWeight:'normal',
  },
  buttonLock: {
    flex: 1,
    backgroundColor: '#FFA500',
    borderRadius: 30,
  },
  buttonLockOwner: {
    flex: 1,
    fontSize: 20,
    color: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    margin: 10,
  }, 
  buttonLockText: {
    flex: 1,
    fontSize: 18,
    color: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
    margin: 10,
  }, 
  time:{
    flex: 1,
    borderColor: '#E6E5F2',
    borderWidth:1,
    borderRadius: 30,
    padding: 10,
    maxWidth: 160,
  }, 
  timeText: {
    flex: 1,
    fontSize: 16,
    color: '#55418E',
    alignItems: 'center',
    justifyContent: 'center',
    fontWeight: 'bold',
  }, 
  timeRecord:{
    flex: 1,
    fontSize: 16,
    color: '#FFA500',
    alignItems: 'center',
    justifyContent: 'center',
    fontWeight: 'normal',
  }, 

  cost:{
    flex: 1,
    borderColor: '#E6E5F2',
    borderWidth:1,
    borderRadius: 30,
    padding: 10,
    maxWidth: 160,
  }, 
  costText: {
    flex: 1,
    fontSize: 16,
    color: '#55418E',
    alignItems: 'center',
    justifyContent: 'center',
    fontWeight: 'bold',
  }, 
  costRecord:{
    flex: 1,
    fontSize: 16,
    color: '#FFA500',
    alignItems: 'center',
    justifyContent: 'center',
    fontWeight: 'normal',
  }, 
  buttonHistory: {
    flex: 1,
    borderRadius: 30,
    borderColor: '#E6E5F2',
    borderWidth:1,
    maxHeight: 50,
  },
  buttonHistoryText: {
    flex: 1,
    fontSize: 15,
    color: '#55418E',
    alignItems: 'center',
    justifyContent: 'center',
    margin: 10,
  }, 

});