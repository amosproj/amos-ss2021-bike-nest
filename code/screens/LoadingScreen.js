import { StatusBar } from 'expo-status-bar';
import React, { useState } from 'react';
import { ImageBackground, ScrollView, Text, View, StyleSheet, Dimensions } from 'react-native';
import Colors from '../styles/Colors';
import { CreateAccountVia3rdParty } from '../components/CreateAccountVia3rdParty';
import { mainStyles } from '../styles/MainStyles';
import global from "../components/GlobalVars";
import JwtDecoder from "../components/JwtDecoder"

export default function LoginScreen({ navigation }) {


    global.authenticationTokenExists().then((exists) => {
        let initialScreen = "Login";

        if (exists === true) {
            global.getAuthenticationToken().then((jwt) => {
                let jwtPayload = JwtDecoder.decode(jwt);
                //console.log("Exp: " + jwtPayload.exp * 1000 + " Actual: " + Date.now());
                if (jwtPayload.exp * 1000 >= Date.now()) {
                    initialScreen = "FindBikeNest";
                    console.log("Valid JWT found.");
                    console.log("initial screen: " + initialScreen);
                    navigation.navigate(initialScreen);
                }
                else{
                    console.log("JWT is not valid.");
                    console.log("initial screen: " + initialScreen);
                    navigation.navigate(initialScreen);
                }
            })
        }
        else {
            console.log("No JWT found.");
            console.log("initial screen: " + initialScreen);
            navigation.navigate(initialScreen);
        }
    });

    return (
        <View style={mainStyles.container}>
            <Text>Loading</Text>
        </View>
    );
}

