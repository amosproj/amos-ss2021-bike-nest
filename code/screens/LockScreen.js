import React from 'react';
import { StyleSheet, Text, View, Image, TouchableOpacity  } from 'react-native';
import { Dimensions } from "react-native";
import { SimpleLineIcons } from '@expo/vector-icons';
// import {styles} from "./styles";

export default function LockScreen({ navigation }) {
    return(
        <View>
            
            <TouchableOpacity style={styles.Icon}>
                <SimpleLineIcons name="lock" size={24} color="black" />
                <Text style={styles.buttonText}> Lock </Text>
            </TouchableOpacity>
            

            <TouchableOpacity onPress={() => alert('go to next screen')} style={styles.LockButton}>
                <Text style={styles.LockButtonText}> Lock </Text>
            </TouchableOpacity>
        </View>

    );
}

const styles = StyleSheet.create({
    Icon:{
        position: 'absolute',
        left: 60,
        top: 197,
        height:128,
        width: 128,
        borderRadius: 64,
        backgroundColor: '#FFA500',
    },


});