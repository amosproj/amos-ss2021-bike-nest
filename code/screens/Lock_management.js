
import React, { useState } from 'react';
import {FlatList,SafeAreaView, StatusBar,StyleSheet,Text, TouchableOpacity, View,} from 'react-native';
import BikeNest_NavigationFooter from '../components/BikeNest_NavigationFooter';
import { mainStyles } from '../styles/MainStyles';
import { Dimensions } from "react-native";
import colors from "../styles/Colors";

const DATA = [
  {
    id: 'bd7acbea-c1b1-46c2-aed5-3ad53abb28ba',
    title: 'Main Door',
  },
  {  
    id: '3ac68afc-c605-48d3-a4f8-fbd91aa97f63',
    title: 'Bike Spot',
  },
];

const Item = ({ item, onPress, style }) => (
  <TouchableOpacity onPress={onPress} style={[styles.item, style]}>
    <Text style={styles.title}>{item.title}</Text>
  </TouchableOpacity>
);

export default function App() {
  const [selectedId, setSelectedId] = useState(null);

  const renderItem = ({ item }) => {
    const backgroundColor = item.id === selectedId ? '#6e3b6e' : '#f9c2ff';

    return <Item item={item} onPress={() => setSelectedId(item.id)} style={{ backgroundColor }} />;
  };

  return (
    <SafeAreaView style={{flex: 1, backgroundColor: colors.UI_Light_4}}>
        <View style={styles.header}>
            <Text style={mainStyles.h1}> Lock Management</Text>
        </View>
        <View style={styles.container}>
             <FlatList
             data={DATA}
             renderItem={renderItem}
             keyExtractor={item => item.id}
             extraData={selectedId}
             />

        </View>
     
      <BikeNest_NavigationFooter/>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
    header: {
    
        marginTop: 20,
        flexDirection: 'row',
        justifyContent: 'space-around',
        paddingHorizontal: 20,
        alignSelf: 'flex-start',
      },

  container: {
    flex: 1,
    marginTop: StatusBar.currentHeight || 0,
    alignItems: 'center',
    justifyContent: 'center',
   
  },
  item: {
    flex: 1,
    //marginTop: Dimensions.get('window').height * 0.02,
    //marginLeft: Dimensions.get('window').width * 0.01,
    //resizeMode: 'contain',
    //backgroundColor: '#fff',
    //alignItems: 'flex-start',
    //justifyContent: 'space-evenly',
    //alignContent: 'space-around',
    alignItems: 'center',
    justifyContent: 'center',
    padding:10,
    marginVertical: 8,
    marginHorizontal: 16,
    width: 328,
    height: 55,
    borderRadius: 38,
    backgroundColor: colors.UI_Light_2
  },
  title: {
    fontSize: 32,
  },
});