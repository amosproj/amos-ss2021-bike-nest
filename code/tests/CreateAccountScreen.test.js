import React from 'react';
import renderer from 'react-test-renderer';
import { CreateAccountScreen } from '../screens/CreateAccountScreen';

it('renders correctly', () => {
  const tree = renderer
    .create(<CreateAccountScreen />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});