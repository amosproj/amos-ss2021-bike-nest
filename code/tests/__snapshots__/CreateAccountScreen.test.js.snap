// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    style={
      Array [
        undefined,
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "flex": 1,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "transparent",
              "marginTop": 40,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "fontSize": 28,
              "fontWeight": "bold",
              "margin": 30,
              "padding": 15,
            }
          }
        >
          Profil erstellen
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "flex": 1,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#4267B2",
              "borderRadius": 38,
              "elevation": 3,
              "height": 55,
              "justifyContent": "center",
              "margin": 9,
              "opacity": 1,
              "padding": 18,
              "width": 328,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "height": 32,
                  "left": 18,
                  "position": "absolute",
                  "top": 0,
                  "width": 32,
                }
              }
            >
              <Image
                source={1}
              />
            </View>
            <Text
              style={
                Object {
                  "color": "#F2F2F2",
                  "fontSize": 14,
                  "height": "100%",
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              MIT FACEBOOK ANMELDEN
            </Text>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderRadius": 38,
              "elevation": 3,
              "height": 55,
              "justifyContent": "center",
              "margin": 9,
              "opacity": 1,
              "padding": 18,
              "width": 328,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "height": 32,
                  "left": 18,
                  "position": "absolute",
                  "top": 0,
                  "width": 32,
                }
              }
            >
              <Image
                source={1}
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#F2F2F2",
                    "fontSize": 14,
                    "height": "100%",
                    "textAlign": "center",
                    "width": "100%",
                  },
                  Object {
                    "color": "#000000",
                  },
                ]
              }
            >
              MIT GOOGLE ANMELDEN
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "flex": 1,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "transparent",
              "margin": 10,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          ODER MIT EMAIL ANMELDEN
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "flex": 1,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <Modal
          animationType="slide"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          transparent={true}
          visible={false}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#00000090",
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "borderRadius": 15,
                  "elevation": 3,
                  "height": 184,
                  "justifyContent": "space-between",
                  "padding": 18,
                  "width": 288,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 18,
                    "fontWeight": "300",
                  }
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 14,
                  }
                }
              >
                
              </Text>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#F28D35",
                    "borderRadius": 38,
                    "elevation": 3,
                    "height": 55,
                    "justifyContent": "center",
                    "margin": 9,
                    "opacity": 1,
                    "padding": 18,
                    "width": 128,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#F2F2F2",
                        "fontSize": 14,
                        "height": "100%",
                        "textAlign": "center",
                        "width": "100%",
                      }
                    }
                  >
                    Ok
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </Modal>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Vorname"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#F2F2F2",
              "borderRadius": 15,
              "color": "#333333",
              "height": 55,
              "margin": 9,
              "paddingLeft": 12,
              "width": 328,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Name"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#F2F2F2",
              "borderRadius": 15,
              "color": "#333333",
              "height": 55,
              "margin": 9,
              "paddingLeft": 12,
              "width": 328,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Email"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "#F2F2F2",
              "borderRadius": 15,
              "color": "#333333",
              "height": 55,
              "margin": 9,
              "paddingLeft": 12,
              "width": 328,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Passwort"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "#F2F2F2",
              "borderRadius": 15,
              "color": "#333333",
              "height": 55,
              "margin": 9,
              "paddingLeft": 12,
              "width": 328,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 20,
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "margin": 8,
              }
            }
          >
            Datenschutzrichtlinien gelesen
          </Text>
          <View
            accessibilityRole="checkbox"
            accessibilityState={
              Object {
                "checked": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "margin": 5,
                "marginLeft": 10,
                "marginRight": 10,
                "opacity": 1,
                "padding": 10,
              }
            }
            testID="checkbox"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F28D35",
              "borderRadius": 38,
              "elevation": 3,
              "height": 55,
              "justifyContent": "center",
              "margin": 9,
              "opacity": 1,
              "padding": 18,
              "width": 328,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#F2F2F2",
                  "fontSize": 14,
                  "height": "100%",
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              Los geht's!
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0.4,
            "justifyContent": "space-between",
            "marginBottom": 65,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#A1A4B2",
            }
          }
        >
          BEREITS REGISTRIERT? 
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": undefined,
                "textDecorationLine": "underline",
              }
            }
          >
            HIER GEHTS ZUM LOGIN
          </Text>
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 1384,
          "width": 750,
        },
        Object {
          "zIndex": -1,
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
